File: examples/kinesis/0_basic/0_create_did.rs
// Copyright 2020-2024 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use examples_kinesis::create_kinesis_did_document;
use examples_kinesis::get_client_and_create_account;

use examples_kinesis::get_memstorage;

/// Demonstrates how to create a DID Document and publish it on chain.
///
/// In this example we connect to a locally running private network, but it can be adapted
/// to run on any IOTA node by setting the network and faucet endpoints.
///
/// See the following instructions on running your own private network
/// https://github.com/iotaledger/hornet/tree/develop/private_tangle
#[tokio::main]
async fn main() -> anyhow::Result<()> {
  // create new client to interact with chain and get funded account with keys
  let storage = get_memstorage()?;
  let identity_client = get_client_and_create_account(&storage).await?;

  // create new DID document and publish it
  let (document, _) = create_kinesis_did_document(&identity_client, &storage).await?;
  println!("Published DID document: {document:#}");

  // check if we can resolve it via client
  let resolved = identity_client.resolve_did(document.id()).await?;
  println!("Resolved DID document: {resolved:#}");

  Ok(())
}

File: bindings/wasm/identity_wasm/examples/src/0_basic/0_create_did.ts
// Copyright 2020-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { IotaDID } from "@iota/identity-wasm/node";
import { IotaClient } from "@iota/iota-sdk/client";
import { createDocumentForNetwork, getFundedClient, getMemstorage, NETWORK_URL } from "../util";

/** Demonstrate how to create a DID Document and publish it. */
export async function createIdentity(): Promise<void> {
    // create new client to connect to IOTA network
    const iotaClient = new IotaClient({ url: NETWORK_URL });
    const network = await iotaClient.getChainIdentifier();

    // create new client that offers identity related functions
    const storage = getMemstorage();
    const identityClient = await getFundedClient(storage);

    // create new unpublished document
    const [unpublished] = await createDocumentForNetwork(storage, network);
    console.log(`Unpublished DID document: ${JSON.stringify(unpublished, null, 2)}`);
    let did: IotaDID;

    console.log("Creating new identity");
    const { output: identity } = await identityClient
        .createIdentity(unpublished)
        .finish()
        .buildAndExecute(identityClient);
    did = identity.didDocument().id();

    // check if we can resolve it via client
    const resolved = await identityClient.resolveDid(did);
    console.log(`Resolved DID document: ${JSON.stringify(resolved, null, 2)}`);
}
